README.md

```diff
@@ -114,6 +114,7 @@ Lazy-loads the rest of the codebase as a module singleton (so it's only ever fet
 | `setContext`                    | N        | `Function` | `() => window.location.href`                                       | Returns a string representing the state of the app during an occurrence; used to directly set payload value `data.context`                                                                                                                                                                                                                                                                                       |
 | `shouldIgnoreOccurrence`        | N        | `Function` | `undefined`                                                        | Returns a boolean value determining whether the currently-processing occurrence should be sent to Rollbar. Function signature: `shouldIgnoreOccurrence(payload, configuration)` where `payload` is the payload object about to be sent to Rollbar and `configuration` is the configuration object                                                                                                                |
 | `userInfo`                      | N        | `Object`   | `undefined`                                                        | User details (**NOTE: only "email", "id", and "username" fields are indexed by Rollbar**; you can send other fields, but they won't be searchable)                                                                                                                                                                                                                                                               |
+| `transform`        | N        | `Function` | `no-op function`                                                        | Enables users to transform the payload that will be sent to Rollbar. The function must mutate the passed payload. Function signature: `transform(payload, configuration)` where `payload` is the payload object about to be sent to Rollbar and `configuration` is the configuration object                                                                                                                |
```

Comment:
Can you explain the difference between this and `customPayloadFields`? They appear very similar. I think we can use what's already there to accomplish this. `customPayloadFields` overwrites payload values in the `report()` function, so if you pass the custom stack trace data to `customPayloadFields.data.body.telemetry.trace.frames` you should get what you're looking for.

I used these docs to figure that out: https://docs.rollbar.com/reference/create-item

Reply:
In fact we're not handling stack traces at all in this library, so we could add that. I could have sworn that was in here but I guess not.

Reply:
The problem with `customPayloadFields` is:
1. We can only add to the payload, not transform the existing payload, so when the original payload already has stack traces, we can't modify them
2. A custom payload would not have access to the error that triggered the Rollbar logging, except if we forced users to create their own error wrapper handlers (including uncaught) and leave it up to them to gather stack traces from there, which doesn't seem super maintainable to me

We do have stack traces in the library, code normalizing stack frames [here](https://github.com/babbel/rollbar-client.js/blob/main/src/RollbarClientSubmitter.mts#L49), used [here](https://github.com/babbel/rollbar-client.js/blob/main/src/RollbarClientSubmitter.mts#L177).

Reply:
OK this makes sense to me now, thanks for the explanation. I knew stack frame support was in there somewhere!

One change that would be nice for consistency's sake would be to show the actual default function (like for `setContext`) instead of saying `no-op function`. So in this case

```suggestion
| `transform`        | N        | `Function` | `() => {}`                                                        | Enables users to transform the payload that will be sent to Rollbar. The function must mutate the passed payload. Function signature: `transform(payload, configuration)` where `payload` is the payload object about to be sent to Rollbar and `configuration` is the configuration object                                                                                                                |
```